workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSSystem.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    [DisableAutoCreation]
+    public abstract class SpatialOSSystem : ComponentSystem
+    {
+        public WorkerBase Worker { get; private set; }
+        public SpatialOSWorld SpatialWorld { get; private set; }
+        public Connection Connection { get; private set; }
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            SpatialWorld = World as SpatialOSWorld;
+            if (SpatialWorld == null)
+            {
+                throw new ArgumentException("This system should only be run when connected to SpatialOS.");
```

Comment:
I think this is slightly inaccurate, because even if they are connected to SpatialOS a user could by error try to add a SpatialOSSystem to a non-SpatialWorld (e.g. when they create their own `World`).